praw/models/comment_forest.py

```diff
@@ -72,12 +73,18 @@ def __len__(self) -> int:
         return len(self._comments)
 
     def _insert_comment(self, comment):
-        assert comment.name not in self._submission._comments_by_id
+        if comment.name in self._submission._comments_by_id:
+            raise MoreCommentException(
+                "You cannot replace MoreComments in a refreshed comment."
```

Comment:
Consider naming this `DuplicateReplaceException`, with the message embedded in the class along the lines of "A duplicate comment has been detected. Are you attempting to call `replace_more_comments` more than once?